### PR TITLE
DOC: Move and expand Loggers docs. Add rel notes.

### PR DESCRIPTION
**Pull request by danielballan**

### CODE REVIEW

File: doc/source/release-notes.rst [hunk 1]
```diff
@@ -2,6 +2,58 @@
...
+Bug Fixes
+---------
+
+* When a new Subscription is added, send the most recent value immediately but
+  only to the *new* Subscription. Previous releases sent redundant messages
+  to *all* Subscriptions that had similar parameters.
+* Reduce the maximum size of a datagram of search requests to match the typical
+  Maximum Transmission Unit seen in the wild.
+* Fix a bug in the pyepics-compatibility layer that caused the connection
+  callbacks never to be called.
```
Comment (klauer) [inline doc/source/release-notes.rst]:
```suggestion
  callbacks never to be called when underlying ``caproto.threading.client.PV``s were reused.
```

File: doc/source/release-notes.rst [hunk 2]
```diff
@@ -2,6 +2,58 @@
...
+* Fix a typo in the PV names in the ``spoof_beamline`` IOC.
+* Never send an ``EventAddResponse`` after a matching ``EventCancelResponse``
+  has been sent.
+* Always send a response to a failed write, and include the correct error code.
+* If a circuit has an oversized backlog of received commands to process, log a
+  WARNING before disconnecting.
+
+Server Perforance Tuning
```
Comment (klauer) [inline doc/source/release-notes.rst]:
```suggestion
Server Performance Tuning
```

File: doc/source/release-notes.rst [hunk 3]
```diff
@@ -2,6 +2,58 @@
...
+* When a ``Read[Notify]Request`` arrives on the heels of a
+  ``Write[Notify]Request``, wait for up to 0.001 seconds for the write to
+  process before reading the current value. If the write happens to complete
+  in less than 0.001 seconds, the read will reflect the new value. This
+  behavior is in the spirit of, but distinct from, EPICS' "synchronous writes."
+  EPICS allows a device to block while writing if it promises to finish quickly
+  (< 100 microseconds). We take a different approach, making all writes
+  asynchronous. This ensures that an accidentally-slow write cannot lock up the
+  server. It adds latency to some reads, up to a hard maximum of 1000
+  microseconds, giving the effect of synchronous write whenever the write
```
Comment (klauer) [inline doc/source/release-notes.rst]: Also known as 1000000 nanoseconds!

Comment (danielballan) [reply doc/source/release-notes.rst]: Alright, alright. This what a get for respectfully using EPICS' units?

**Pull request merged**